### PR TITLE
Add BC7_SRGB format support

### LINKED ISSUE #277: BC7 UNORM SRGB support (KTX2)
Why is only BC7 UNORM supported?

### PR DESCRIPTION
**Pull request by w23**
Tracked alongside regular BC7 for the most part, with simple conversion from/to external BC7 UNORM/SRGB format enums. No actual sRGB/gamma↔linear colorspace conversion is introduced.

Fixes #277

### CODE REVIEW

File: applications/_plugins/common/texture.h [hunk 1]
```diff
@@ -107,6 +107,7 @@ extern "C" {
 
 #define CMP_FOURCC_BC6H                    CMP_MAKEFOURCC('B', 'C', '6', 'H')
 #define CMP_FOURCC_BC7                     CMP_MAKEFOURCC('B', 'C', '7', 'x')
+#define CMP_FOURCC_BC7_SRGB                CMP_MAKEFOURCC('B', 'C', '7', 's')
```
Comment (w23) [inline applications/_plugins/common/texture.h:110]: Is this a reasonable FourCC? Does it have to be congruent to a well-known value?

**Comment by w23** [comment]
#114 mentions that there's supposedly an SRGB flag, but I haven't found one.

File: applications/_plugins/common/texture.h (hunk 1 above)
Comment (NPCompress) [reply applications/_plugins/common/texture.h:110]: @w23  This seems reasonable, I am not aware of any other open source options that use it.

**Comment by NPCompress** [comment]
@SlawekNowy Please review your latest pull request against v4.5 there are some conflicting files that need resolving. Thx

**Comment by NavNTCMP** [comment]
@w23  Just a reminder to review your latest pull request against v4.5.52 source code. Thx for your support.

**Comment by w23** [comment]
Master branch has been blindly and mechanically merged. Note that I haven't tested it yet at all. It's been quite a while since the last time I needed it and had a proper build+test setup. Probably will have to fixup my KTX2-support-for-Linux patchset first.

Also note that, this PR doesn't add any BC7_SRGB support neither to GUI, nor to CLI args. It just adds format tracking, which seems to be enough when re-packing images from one container to another.

There are also no tests added -- I haven't yet figured out how to build and run them.

**Comment by kaphula** [comment]
Sorry if this is the wrong place to ask but what is the current status for supporting conversions from png to ktx2 with BC1, BC3, BC5 and BC7 compression formats on Linux? It is seems that currently compressonator does not allow me to convert to any kind of ktx2 files on Linux, but on Windows conversions to ktx2 are possible?

**Comment by w23** [comment]
> Sorry if this is the wrong place to ask but what is the current status for supporting conversions from png to ktx2 with BC1, BC3, BC5 and BC7 compression formats on Linux?

Last time I looked at it (in December), KTX2 support was force-disabled for Linux at compile time.
Good news: it is mostly a repository-structure + build-system issue. Basic changes needed to enable KTX2 support on Linux are fairly small: https://github.com/GPUOpen-Tools/compressonator/compare/master...w23:compressonator:linux-ktx2
Bad news: it is mostly a repository-structure + build-system issue. Massaging the changes linked above so they conform to (completely opaque) AMD internal CI build pipelines so that a realistically-mergeable PR can be prepared might require much more coordination than I have resources for.

> It is seems that currently compressonator does not allow me to convert to any kind of ktx2 files on Linux, but on Windows conversions to ktx2 are possible?

I heard KTX2 _just works&trade;_ on Windows.

**Comment by kaphula** [comment]
Thanks for the summary. I guess I will try running Compressonator on Linux through Wine and hope for official future ktx2 support on Linux.

**Pull request merged**